src/isomorphic/children/sliceChildren.js

```diff
@@ -29,23 +27,8 @@ function sliceChildren(children, start, end) {
     return children;
   }
 
-  var slicedChildren = {};
-  var flattenedMap = flattenChildren(children);
-  var ii = 0;
-  for (var key in flattenedMap) {
-    if (!flattenedMap.hasOwnProperty(key)) {
-      continue;
-    }
-    var child = flattenedMap[key];
-    if (ii >= start) {
-      slicedChildren[key] = child;
-    }
-    ii++;
-    if (end != null && ii >= end) {
-      break;
-    }
-  }
-  return ReactFragment.create(slicedChildren);
+  var array = ReactChildren.toArray(children);
+  return array.slice(start, end);
```

Comment:
Just noting that this behavior should be unchanged because both `traverseAllChildren` and `toArray` excludes `null`.